### ISSUE TITLE
Not published units are visible on the course outline

### ISSUE THREAD

**Issue by fdns** [regisb/tutor#128]
When students access a course with a non-published unit, they can see it. 

This is different from a native installation, where the unit is not visible, which make me think we are missing a setting or configuration. I have looked at the MODULESTORE and MODULESTORE_BRANCH differences with the native installation, but they are the same (I also tried diffing all the settings, and have not found anything useful).

## Steps to reproduce
Tested commit: 49317ac3dfabcd875f8c731491714c0db12e8039

* build and run tutor.
* Create course
* Add a section.
* Add a subsection.
* Add a unit (don't publish it).
* Set the course to Self-Paced Courses.
* Set the course to released.
* Access the course as a student.

## Unexpected behavior

As a normal user, you shouldn't be able to see the name of the unpublished data.

## Additional info
Here are some images with the behavior I'm seeing:

Native installation unpublished unit (course released): https://imgur.com/KDEXN75
Tutor unpublished unit (course released): https://imgur.com/DSEfk4V https://imgur.com/2PaO74a

**Comment by regisb** [comment]
Hi @fdns! Thanks for the detailed report, I managed to reproduce the issue. However, trying to access the unpublished unit raises a "Page not found" error. That means that students should not be able to access the content of the unpublished unit. Still, this is inconvenient both for students and course designers.

I looked at every setting in aws.py and common.py and did not find a clue there. Next step is to run the native Open edX devstack to compare its behaviour with tutor.
